File: Sully.c
#include <stdio.h>
#include <stdlib.h>
int i = 5;
char str[5000];
char str2[5000];
int main()
{
if (i == 1)
return (0);
i--;
char *quine = "#include <stdio.h>%c#include <stdlib.h>%cint i = %d;%cchar str[5000];%cchar str2[5000];%cint main()%c{%cif (i == 1)%creturn (0);%ci--;%cchar *quine = %c%s%c;%csprintf(str, %cSully_%%d.c%c, i);%csprintf(str2, %cSully_%%d%c, i);%cFILE *file = fopen(str, %cw%c);%cfprintf(file, quine, 10, 10, i, 10, 10, 10, 10, 10, 10, 10, 10, 34, quine, 34, 10, 34, 34, 10, 34, 34, 10, 34, 34, 10, 10, 10, 10, 34, 34, 10, 10, 10, 34, 34, 10, 10, 10);%cfclose(file);%cchar str3[5000];%csprintf(str3, %cgcc -o %%s %%s%c, str2, str);%csystem(str3);%cchar str4[5000];%csprintf(str4, %c./%%s%c, str2);%csystem(str4);%c}%c";
sprintf(str, "Sully_%d.c", i);
sprintf(str2, "Sully_%d", i);
FILE *file = fopen(str, "w");
fprintf(file, quine, 10, 10, i, 10, 10, 10, 10, 10, 10, 10, 10, 34, quine, 34, 10, 34, 34, 10, 34, 34, 10, 34, 34, 10, 10, 10, 10, 34, 34, 10, 10, 10, 34, 34, 10, 10, 10);
fclose(file);
char str3[5000];
sprintf(str3, "gcc -o %s %s", str2, str);
system(str3);
char str4[5000];
sprintf(str4, "./%s", str2);
system(str4);
}

File: bonus/Sully.php
<?php
$i = 5;
if ($i == 0)
return (0);
if (strstr($argv[0], "_") != false)
$i = $i - 1;
$quine = '<?php%c$i = %d;%cif ($i == 0)%creturn (0);%cif (strstr($argv[0], "_") != false)%c$i = $i - 1;%c$quine = %c%s%c;%c$handle = fopen("Sully_".$i.".php", "w");%cfprintf($handle, $quine, 10, $i, 10, 10, 10, 10, 10, 39, $quine, 39, 10, 10, 10, 10, 10, 10);%cfclose($handle);%cshell_exec("php Sully_".$i.".php");%c?>%c';
$handle = fopen("Sully_".$i.".php", "w");
fprintf($handle, $quine, 10, $i, 10, 10, 10, 10, 10, 39, $quine, 39, 10, 10, 10, 10, 10, 10);
fclose($handle);
shell_exec("php Sully_".$i.".php");
?>
